### PR TITLE
chore: [BBND-469] npm workspaces

### PR DESCRIPTION
**Pull request by ManuIOB**
**Description**:

Add NPM workspaces configuration

**Related issue(s)**:

Fixes #

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by mishomihov00 — CHANGES_REQUESTED** [review]
All workflows seems to be failing on the `npm ci` command due to a missing `package-lock.json` file. Please check.

File: package.json [hunk 1]
```diff
@@ -54,14 +54,21 @@
...
     "serverless": "^3.34.0"
   },
   "devDependencies": {
     "@commitlint/cli": "^19.2.0",
     "@commitlint/config-conventional": "^19.1.0",
     "dotenv": "^16.3.1",
     "fs": "^0.0.1-security",
-    "husky": "^9.0.11"
-  }
+    "husky": "^9.0.11",
```
[the remark this answers is not in the document]
Comment (MiguelLZPF) [reply package.json:64]: "shared dependencies declared in the root package.json are installed once at the root and used by all packages that require them, avoiding duplication in sub-packages"

File: package.json
[the remark this answers is not in the document]
Comment (ManuIOB) [reply package.json:1]: added, thanks

File: package.json [hunk 2]
```diff
@@ -6,8 +6,8 @@
   "author": "",
   "license": "Apache-2.0",
   "scripts": {
-    "setup": "npm run install",
-    "install": "node install.js && npm run build:cli:full && npm run prepare",
+    "setup": "npm run install:all",
+    "install:all": "npm install && npm run build:cli:full && npm run prepare",
```
[the remark this answers is not in the document]
Comment (ManuIOB) [reply package.json]: updated package.json scripts

File: package.json [hunk 3]
```diff
@@ -6,8 +6,8 @@
...
     "install:backend": "cd backend && npm install",
     "install:contracts": "cd contracts && npm install",
```
[the remark this answers is not in the document]
Comment (ManuIOB) [reply package.json]: updated package.json scripts

File: package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (MiguelLZPF) [reply package.json]: This one can also be addressed in future PRs if it is a lot of work right now... 👍🏻

[the remark this answers is not in the document]
Comment (ManuIOB) [reply package.json]: true, i had issues about that, but I found a solution `"prepare": "husky install --workspace=contracts"`

**Review by MiguelLZPF — APPROVED** [review]

**Pull request merged**